cmd/root.go

```diff
@@ -1,61 +1,663 @@
+			return err
+		}
+
+		log.Debugf("Successfully updated JIRA issue %s!", jIssue.Key)
+	} else {
+		log.Debugf("JIRA issue %s is already up to date!", jIssue.Key)
+	}
+
+	issue, _, err := jClient.Issue.Get(jIssue.ID, nil)
+	if err != nil {
+		log.Errorf("Error retrieving JIRA issue %s to get comments.", jIssue.Key)
+	}
+
+	var comments []jira.Comment
+	if issue.Fields.Comments == nil {
+		log.Debugf("JIRA issue %s has no comments.", jIssue.Key)
+		comments = make([]jira.Comment, 0)
+	} else {
+		commentPtrs := issue.Fields.Comments.Comments
+		comments = make([]jira.Comment, len(commentPtrs))
+		for i, v := range commentPtrs {
+			comments[i] = *v
+		}
+		log.Debugf("JIRA issue %s has %d comments", jIssue.Key, len(comments))
+	}
+
+	if err = createComments(ghIssue, jIssue, comments, ghClient, jClient); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createIssue(issue github.Issue, ghClient github.Client, jClient jira.Client) error {
+	log.Debugf("Creating JIRA issue based on GitHub issue #%d", *issue.Number)
+
+	fields := jira.IssueFields{
+		Type: jira.IssueType{
+			Name: "Task", // TODO: Determine issue type
+		},
+		Project:     project,
+		Summary:     *issue.Title,
+		Description: *issue.Body,
+		Unknowns:    map[string]interface{}{},
+	}
+
+	key := fmt.Sprintf("customfield_%s", ghIDFieldID)
+	fields.Unknowns[key] = *issue.ID
+	key = fmt.Sprintf("customfield_%s", ghNumFieldID)
+	fields.Unknowns[key] = *issue.Number
+	key = fmt.Sprintf("customfield_%s", ghStatusFieldID)
+	fields.Unknowns[key] = *issue.State
+	key = fmt.Sprintf("customfield_%s", ghReporterFieldID)
+	fields.Unknowns[key] = issue.User.GetLogin()
+	key = fmt.Sprintf("customfield_%s", ghLabelsFieldID)
+	strs := make([]string, len(issue.Labels))
+	for i, v := range issue.Labels {
+		strs[i] = *v.Name
+	}
+	fields.Unknowns[key] = strings.Join(strs, ",")
+	key = fmt.Sprintf("customfield_%s", isLastUpdateFieldID)
+	fields.Unknowns[key] = time.Now().Format(dateFormat)
+
+	jIssue := &jira.Issue{
+		Fields: &fields,
+	}
+
+	jIssue, res, err := jClient.Issue.Create(jIssue)
+	if err != nil {
+		log.Errorf("Error creating JIRA issue: %s", err)
+		defer res.Body.Close()
+		body, _ := ioutil.ReadAll(res.Body)
+		log.Debugf("Error body: %s", body)
+		return err
+	}
+
+	log.Debugf("Created JIRA issue %s!", jIssue.Key)
+
+	if err = createComments(issue, *jIssue, nil, ghClient, jClient); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var jCommentRegex = regexp.MustCompile("^Comment \\(ID (\\d+)\\) from GitHub user (\\w+) \\((.+)\\)? at (.+):\\n\\n(.+)$")
+var jCommentIDRegex = regexp.MustCompile("^Comment \\(ID (\\d+)\\)")
+
+func createComments(ghIssue github.Issue, jIssue jira.Issue, existing []jira.Comment, ghClient github.Client, jClient jira.Client) error {
+	if *ghIssue.Comments == 0 {
+		log.Debugf("Issue #%d has no comments, skipping.", *ghIssue.Number)
+		return nil
+	}
+
+	ctx := context.Background()
+	repo := strings.Split(rootCmdCfg.GetString("repo-name"), "/")
```

Comment:
you do this same parsing in multiple places. `repo` should be a field on a config object so it can be reused.